### DISCUSSION TITLE
Fork Tools settings are reset after fork container is restarted

### DISCUSSION THREAD

**Discussion by LeroyINC** [guydavis/machinaris#681] [General]
if I edit the config.yaml or use the for tools to change a setting.. then start\top the docker container the setting gets erased.
should this setting be saved somewhere?

the specific setting I change is:
SETFNTARGETPEERCOUNT='20'

I usually increase it to a higher number like 80

Note: i am not doing anything else to the container like recreating or upgrading -- only a restart

**Comment by guydavis** [comment]
Sorry, I'm not able to replicate this for a fork blockchain such as Cactus.   Changing the Settings | Tools value as shown: 
![image](https://user-images.githubusercontent.com/12936/171682179-6dcdd2bb-79e2-4e0f-9f3b-4bba403f196d.png)

updated: `/root/.chia/forktools/ftcconfigs/config.forkfixconfig` which is persisted on disk. This resulted in the `target_peer_count` value being automatically changed in my config.yaml and the blockchain farming restarted.

![image](https://user-images.githubusercontent.com/12936/171682364-e0093d7e-7932-48f3-9308-b4cc6cf7e05b.png)

I then restarted the entire Machinaris container (Cactus in my testing) and the value persisted.

Are we perhaps talking only about Chia (now version 1.3)?   If so, please check the contents of `/tmp/forktools_setup.log` in the container for details of the application on container launch.  Settings behavior should be the same. Hope this helps!

**Comment by LeroyINC** [comment]
In my case it was Silicoin and Hddcoin

Thank for the effort on this



Leroy Ladyzhensky
Integrated Network Concepts
440-937-7680

Sent from my Smart Phone





-------- Original message --------
From: Guy Davis ***@***.***>
Date: 6/2/22 12:57 PM (GMT-05:00)
To: guydavis/machinaris ***@***.***>
Cc: "Leroy L. Ladyzhensky" ***@***.***>, Author ***@***.***>
Subject: Re: [guydavis/machinaris] Fork Tools settings are reset after fork container is restarted (Discussion #681)


Sorry, I'm not able to replicate this for a fork blockchain such as Cactus. Changing the Settings | Tools value as shown:
[image]<https://user-images.githubusercontent.com/12936/171682179-6dcdd2bb-79e2-4e0f-9f3b-4bba403f196d.png>

updated: /root/.chia/forktools/ftcconfigs/config.forkfixconfig which is persisted on disk. This resulted in the target_peer_count value being automatically changed in my config.yaml and the blockchain farming restarted.

[image]<https://user-images.githubusercontent.com/12936/171682364-e0093d7e-7932-48f3-9308-b4cc6cf7e05b.png>

I then restarted the entire Machinaris container (Cactus in my testing) and the value persisted.

Are we perhaps talking only about Chia (now version 1.3)? If so, please check the contents of /tmp/forktools_setup.log in the container for details of the application on container launch. Settings behavior should be the same. Hope this helps!

**Comment by LeroyINC** [comment]
looks like my setting were reset again... here is the log..

from the log it looks like it changed it from 80 back to 20

I did restart this container at that time indicated in the log

----- log

fatal: destination path 'forktools' already exists and is not an empty directory.
Installing forktools main branch...
Updating ( git fetch --all, git reset --hard origin/main ) to most current version of forktools main branch...
Fetching origin
HEAD is now at 5df93f7 Merge pull request #22 from Qwinn1/testing
Curl installed.  Proceeding with forktools installation.
Setting up environment variables in /root/.bashrc
Scanning for and setting up required symlinks for forks with non-standard paths...
Copying config.FORKTOOL.template files to config.FORKTOOL if necessary...
No existing config.forkstartall file found.  Copied from config.forkstartall.template.
  WARNING:  forkstartall will not function until config.forkstartall is manually configured.
No existing config.forkaddplotdirs file found.  Copied from config.forkaddplotdirs.template.
  WARNING:  forkaddplotdirs will not function until config.forkaddplotdirs is manually configured.
No existing config.forkremoveplotdirs file found.  Copied from config.forkremoveplotdirs.template.
  WARNING:  forkremoveplotdirs will not function until config.forkremoveplotdirs is manually configured.
No existing config.forklog file found.  Copied from config.forklog.template.
  forklog will function correctly with forktools defaults, but user may change defaults as desired in config.forklog .
No existing config.forkexplore file found.  Copied from config.forkexplore.template.
  forkexplore will function correctly with forktools defaults, but user may change defaults as desired in config.forkexplore .
No existing config.forkfixconfig file found.  Copied from config.forkfixconfig.template.
  forkfixconfig will function correctly with forktools defaults, but user may change defaults as desired in config.forkfixconfig .
No existing config.forkedit file found.  Copied from config.forkedit.template.
  forkedit will use gedit as the text editor by default, but this can be changed to your preferred editor in config.forkedit.
No existing config.logging file found.  Copied from config.logging.template.
  Only forkmon logging is enabled by default.  Update config.logging to enable logs for any or every forktool (except forkconfig).
Making forktool scripts executable...
forktools installation completed!  Happy farming!  - Qwinn
HEAD is now at 5df93f7 Merge pull request #22 from Qwinn1/testing
'forkfixconfig all' initiated on Sun Jun 12 15:08:38 EDT 2022...


forkfixconfig btcgreen:  Analyzing /root/.btcgreen/mainnet/config/config.yaml.
Proposed changes to /root/.btcgreen/mainnet/config/config.yaml :
  Old Target Peer Count:  target_peer_count: 80
  New Target Peer Count:  target_peer_count: 20
Should you proceed, a backup of your current config.yaml will be made called config.yaml.2022-06-12

Setting target peer count...
Backed up original btcgreen config.yaml to config.yaml.2022-06-12.  /root/.btcgreen/mainnet/config/config.yaml has had the proposed changes applied.

'forkfixconfig all' complete.  Remember to restart services ('forkstart all -s 30', as an example, to get a 30 second delay between restarts) for any changes to take effect.
'forkpatch all -multiproc' initiated on Sun Jun 12 15:12:43 EDT 2022...


forkpatch:  Analyzing suitability of btcgreen for automated -multiproc patching.
forkpatch:  btcgreen failed code validation - num_workers code missing or has non-standard indendation.  Skipping btcgreen.
'forkpatch all -logwinningplots' initiated on Sun Jun 12 15:19:50 EDT 2022...


forkpatch:  Analyzing suitability of btcgreen for automated -logwinningplots patching.
forkpatch:  btcgreen failed code validation - already patched.  Skipping btcgreen.
~

**Reply by guydavis** [reply]
Hmm, try setting a `SETFNTARGETPEERCOUNT` of 81 instead of 80 on the Settings Tools page.  Looks like the default (80) is being overwritten on launch.  Let me know how it goes.  Thanks.

Update: I dug deeper into the forktools setup code and have [added a guard](https://github.com/guydavis/machinaris/commit/6392c2041fa41ea6960b4fa61eda6058d2a9ee33) to not replace the default value more than once on very initial startup of Machinaris.  So, in your case this would be one more time using the `:develop` image.   Sorry for the trouble and thanks for the report.

**Comment by LeroyINC** [comment]
That seemed to work.. setting the "SETFNTARGETPEERCOUNT" to anything but 80 maintains the setting through docker restarts.

when its set to 80 some containers get changes to 20 at restart
